core/network/src/protocol/sync/extra_requests.rs

```diff
@@ -138,7 +138,7 @@ impl<B: BlockT> ExtraRequests<B> {
 		}
 
 		if best_finalized_number > self.best_seen_finalized_number {
-			match self.tree.finalize_with_ancestors(
+			match self.tree.finalize(
```

Comment:
Could you, please, change it back to `finalize_with_ancestors()` for finality proofs? It isn't guaranteed that they're coming in-order. If that PR needs to be merged ASAP, just ping me - I'll create a follow-up PR.

Reply:
@andresilva @svyatonik i guess this is on-ice until this change is made?

Reply:
@svyatonik Doesn't it have the same guarantees as for justifications? The code that makes the requests is shared with justifications and it will request things in-order as they are "popped" from the tree. I understand that the finality proof for `B` might have a justification for block `B+n`, but for checking in the tree we will be using in `B` anyway right?

Reply:
@andresilva You're right, `on_block_finalized` is called for every block in finalization path => they'll be in-order anyway. But then probably we should reconsider [this limitation](https://github.com/paritytech/substrate/blob/9b08e7ff938a45dbec7fcdb854063202e2b0cb48/core/client/src/client.rs#L1149) - it could happen that we'll only receive notification for `B+n`, but no notification for `B` => we could end up with the same `UnfinalizedAncestor` error, right?

Reply:
Indeed, I did not remember that. Let's revert back to `finalize_with_ancestors()` then.